### PR TITLE
net-libs/libsrtp: 1.6.0-r1 OpenSSL 1.1 compatibility

### PR DESCRIPTION
**Pull request by candrews**
Uses the patches from FreeBSD, see
https://svnweb.freebsd.org/ports?view=revision&revision=472170

Closes: https://bugs.gentoo.org/666160
Package-Manager: Portage-2.3.52, Repoman-2.3.12
Signed-off-by: Craig Andrews <candrews@gentoo.org>

### CODE REVIEW

**Comment by candrews** [comment]
@gentoochainsaw Can you please take a look? Let me know if I can merge this, thanks

File: net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+@@ -76,21 +74,43 @@ hmac_alloc (auth_t **a, int key_len, int out_len)
+         return err_status_bad_param;
+     }
+ 
+-    /* allocate memory for auth and HMAC_CTX structures */
+-    pointer = (uint8_t*)crypto_alloc(sizeof(HMAC_CTX) + sizeof(auth_t));
+-    if (pointer == NULL) {
++/* OpenSSL 1.1.0 made HMAC_CTX an opaque structure, which must be allocated
++   using HMAC_CTX_new.  But this function doesn't exist in OpenSSL 1.0.x. */
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
```
Comment (steils) [inline net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: `#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)`

File: net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+-    new_hmac_ctx = (HMAC_CTX*)((*a)->state);
+-    HMAC_CTX_init(new_hmac_ctx);
+ 
+     /* increment global count of all hmac uses */
+     hmac.ref_count++;
+@@ -106,11 +126,19 @@ hmac_dealloc (auth_t *a)
+ 
+     hmac_ctx = (HMAC_CTX*)a->state;
+ 
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
```
Comment (steils) [inline net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: `#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)`

File: net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+ /*
+  * sha1_init(&ctx) initializes the SHA1 context ctx
+  *
+@@ -72,6 +70,12 @@ typedef EVP_MD_CTX sha1_ctx_t;
+  *
+  */
+ 
++/* OpenSSL 1.1.0 made EVP_MD_CTX an opaque structure, which must be allocated
++   using EVP_MD_CTX_new. But this function doesn't exist in OpenSSL 1.0.x. */
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
```
Comment (steils) [inline net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: `#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)`

Comment (candrews) [reply net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: :+1:

File: net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch (hunk 2 above)
Comment (candrews) [reply net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: :+1:

File: net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch (hunk 1 above)
Comment (candrews) [reply net-libs/libsrtp/files/libsrtp-1.6.0-openssl-1.1.patch]: :+1:

**Comment by candrews** [comment]
@steils Thanks for that - I've updated the patch in this PR as you've suggested.

File: net-libs/libsrtp/libsrtp-1.6.0.ebuild [hunk 4]
```diff
@@ -27,7 +27,7 @@ DOCS=( CHANGES README TODO )
 MULTILIB_WRAPPED_HEADERS=(
 	/usr/include/srtp/config.h
 )
-PATCHES=( "${FILESDIR}/${PN}-pcap-automagic-r0.patch" )
+PATCHES=( "${FILESDIR}/${PN}-pcap-automagic-r0.patch" "${FILESDIR}/${P}-openssl-1.1.patch" )
```
Comment (steils) [inline net-libs/libsrtp/libsrtp-1.6.0.ebuild]: Are you sure you want to add a patch to a stable ebuild in the same revision? Bear in mind that >=dev-libs/openssl-1.1 (for which this patch is made) is still not stable and even masked.

Comment (candrews) [reply net-libs/libsrtp/libsrtp-1.6.0.ebuild]: That was a mistake. My apologies for the trouble!

**Pull request closed without merging**